bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -724,24 +724,25 @@ public Image(Device device, ImageFileNameProvider imageFileNameProvider) {
 public Image(Device device, ImageDataProvider imageDataProvider) {
 	super(device);
 	this.imageDataProvider = imageDataProvider;
-	currentDeviceZoom = DPIUtil.getDeviceZoom ();
-	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, currentDeviceZoom);
+	currentNativeZoom = DPIUtil.getNativeDeviceZoom();
+	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, getZoom());
 	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), data.zoom());
 	init (resizedData);
 	init();
 }
 
 /**
- * Update zoom and refresh the Image based on the zoom level, if required.
+ * Update zoom and refresh the Image based on the native zoom level, if required.
  *
- * @param deviceZoom zoom in % of the standard resolution the image shall be scaled for
+ * @param nativeZoom native zoom in % of the monitor on which the image is painted
  *
  * @return true if image is refreshed
  */
-boolean handleDPIChange (int deviceZoom) {
+boolean handleDPIChange (int nativeZoom) {
+	int deviceZoom = DPIUtil.getZoomForAutoscaleProperty(nativeZoom);
 	boolean refreshed = false;
 	if (imageFileNameProvider != null) {
-		if (deviceZoom != currentDeviceZoom) {
+		if (deviceZoom != getZoom()) {
```

Comment:
Is `deviceZoom` correct here? It's not the zoom of the device but of a monitor, isn't is? Maybe you could also change `deviceZoom` to `newZoom` and assign `getZoom()` to some `oldZoom` variable to make it easier to understand what happens here.

Reply:
no we calculate the device zoom from native zoom at line no. 742
and getZoom gets the devicezoom calculated from the currentNativeZoom.